rfd/0083-license-expiration-warnings.md

```diff
@@ -0,0 +1,144 @@
+---
+authors: edwarddowling (edward.dolwing@goteleport.com), r0mant (roman@goteleport.com)
+---
+
+
+# RFD 83 - License warnings
+
+## Required approvers
+
+- Engineering: `@r0mant`
+- Product: `@klizentas && @xinding33`
+- Security: `@reedloden`
+
+## What
+
+This RFD proposes a way to display license warnings to self hosted users both before and after license expiration.
+
+## Why
+
+To encourage the renewal of licenses after expiration and let users know before their license expires.
+
+## Success criteria
+
+Teleport displays license warnings to users in both CLI and web UI starting from a certain time prior to the license expiration, as well as post-expiration.
+Teleport administrators can optionally configure the license expiration warning interval but not disable it completely on a per role basis.
```

Comment:
@xinding33 To piggyback on this - is having a configurable license warning interval per-role a hard requirement from your standpoint? It's fine if it is - just wanted to double-check since having the same warning interval for everyone would definitely simplify implementation and it's another feature to support.

Also, presumably cluster admins can just set it to the minimum of 10 days for everyone to "suppress" the default behavior which would sort of alleviate our attempts to make it more/less annoying for different types of users.

Reply:
@r0mant Configurable is nice to have but not a hard requirement. As discussed, it's hard to determine who "admins" are and who "regular users" are so that's why we suggested a role option instead of tying it arbitrarily to existing role options or roles.

Are you suggesting we remove the role option entirely and just alert everyone N days ahead? Or are you suggesting we change the role option from integer to boolean?

Reply:
@xinding33 The former:

> remove the role option entirely and just alert everyone N days ahead

If that works for you, that would simplify the implementation significantly and allow us to reuse the same cluster alerts mechanism we'll be using in automatic upgrade notifications.